### PR TITLE
metadisp: initial commit

### PR DESCRIPTION
**Pull request by sheenobu**
Summary:

feature/metadisp is an xlator for performing "metadata dispersal" across
multiple children. it does this by flattening the complex
POSIX paths into /$GFID style paths, then forwarding the
metadata operations to its first child and forwarding the
data operations to its second child.

The purpose of this xlator is to allow separation of data and metadata,
in cases where metadata might be stored in another format (embedded kv?),
on another disk (ssd), on another host (dht2).

### CODE REVIEW

File: tests/basic/metadisp/metadisp.t [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+TEST touch $M0/$FALLOC_FILE
+TEST fallocate -l $((4096*5)) $M0/$FALLOC_FILE
+TEST fallocate -p -o 4096 -l 4096 $M0/$FALLOC_FILE
+# This actually fails with "operation not supported" on most filesystems, so
+# don't leave it enabled except to test changes.
+#EST fallocate -z -o $((4096*3)) -l 4096 $M0/$FALLOC_FILE
+
+logdir=`gluster --print-logdir`
+
+#cleanup;
```
Comment (amarts) [inline tests/basic/metadisp/metadisp.t:294]: Appreciate your effort on getting a nice test case done.

Normally, I try to use a copy of 'rpc-coverage.t' to validate most of the basic fops are working. This is very nice.

**Review by amarts — COMMENTED** [review]
still checking. thanks for the PR.

**Review by amarts — COMMENTED** [review]
couldn't review completely, but a top level review shows this is what I was thinking about. This has most of the factors required already! Awesome. 

@kotreshhr this is a good set  to improve upon :-)

**Comment by sheenobu** [comment]
closing PR for gerrit version https://review.gluster.org/#/c/glusterfs/+/24071

**Pull request closed without merging**